app/utils/validators/forms/FieldValidator.scala

```diff
@@ -115,10 +115,15 @@ object FieldValidator {
   val validateDecimal: Int => Int => String => Boolean = (totalLength: Int) =>
     (decimalPlaces: Int) =>
       (input: String) =>
-        input.split('.') match {
-          case Array(a, b) if isNumeric(a) && isNumeric(b) => b.length <= decimalPlaces && (a + b).length <= totalLength && b.toDouble > 0
-          case Array(a) if isNumeric(a) && nonEmpty(a)     => a.length <= totalLength && a.toDouble > 0
-          case _                                           => false
+        input.trim().endsWith(".") match {
+          case false =>
+            input.trim().split('.') match {
+              case Array(a, b) if isNumeric(a) && isNumeric(b) =>
+                b.length <= decimalPlaces && (a + b).length <= totalLength && (b.toDouble > 0 || a.toDouble > 0)
+              case Array(a) if isNumeric(a) && nonEmpty(a) => a.length <= totalLength && a.toDouble > 0
+              case _                                       => false
+            }
+          case true => false
```

Comment:
Using `true` `false` in pattern matching is not a good approach. In that case `if` is better.

Additionally I think we shouldn't trim value for validating. We're trimming this just for the validation actually, but the original value can have the spaces and fail on the xml validation, because in practice we can have more characters than allowed.

Moreover this validation is really complex and cases like `case true => false` are not enough described to understand it.

I'll come back to totally first implementation for `validateDecimal` or go with @PauloPortugal idea to validate it using `BigDecimal`

Reply:
Additionally `0` is a correct `BigDecimal`. This file contains general validation helper. We should add validation for zero just in specific case where we want to use it.

Reply:
Done please review